chispa/default_formats.py

```diff
@@ -2,8 +1,67 @@
+from __future__ import annotations
+
+import warnings
+from dataclasses import dataclass, field
+from typing import Any
+
+from chispa.formatting import Color, Format, FormattingConfig, Style
 
 
 @dataclass
 class DefaultFormats:
-    mismatched_rows = ["red"]
-    matched_rows = ["blue"]
-    mismatched_cells = ["red", "underline"]
-    matched_cells = ["blue"]
+    """
+    This class is now deprecated and should be removed in a future release, together with `convert_to_formatting_config`.
+    """
+
+    mismatched_rows: list[str] = field(default_factory=lambda: ["red"])
+    matched_rows: list[str] = field(default_factory=lambda: ["blue"])
+    mismatched_cells: list[str] = field(default_factory=lambda: ["red", "underline"])
+    matched_cells: list[str] = field(default_factory=lambda: ["blue"])
+
+    def __post_init__(self):
+        warnings.warn(
+            "DefaultFormats is deprecated. Use `chispa.formatting.FormattingConfig` instead.", DeprecationWarning
+        )
+
+
+def convert_to_formatting_config(instance: Any) -> FormattingConfig:
```

Comment:
I would like to have it as static method of `FormattingConfig`. Maybe even private one.

Reply:
Good suggestion, I created `FormattingConfig._from_arbitrary_dataclass`